enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,502 @@
+    // +kubebuilder:default:="Unknown"
+    // +optional
+    ProviderName string `json:"providerName,omitempty"`
+}
+
+type PlatformStatus struct {
+    ...
+    // External contains settings specific to the generic External infrastructure provider.
+    // +optional
+    External *ExternalPlatformStatus `json:"external,omitempty"`
+}
+```
+
+### Implementation Details/Notes/Constraints
+
+#### Implementation phases
+
+Splitting the project into phases would be natural to make the implementation process smoother.
+A reader can find the proposed phase breakdown in [OCPPLAN-9429](https://issues.redhat.com/browse/OCPPLAN-9429).
+
+This document intends to describe the initial phases of this project. The proposed initial course of action:
+
+1. Update "openshift/api" with adding "External" PlatformType
+2. Ensure that all Red Hat operators tolerate the "External" platform and treat the same as the "None" platform
+
+Next phase out of the scope for this EP:
+
+3. Modify operators for doing specific things when seeing the "External" platform.
+   It was described briefly in the [Affected Components](#affected-components) section.
+   However, this should be addressed in separate EPs on a per-component basis.
+
+#### Future work
+
+There are several topics in this area that would be wise to defer for upcoming phases, namely:
+
+1. Define missing capabilities and their concrete behaviour, for example, add a "capability" for machine-api
+2. Precisely define the reaction of the operators listed below for the "External" platform type
+3. Define and document concrete mechanism for supplement cluster with provider-specific components during installation time (CCM, MAPI controller, DNS controller)
+
+### Risks and Mitigations
+
+#### Support burden
+There is concern that *random* customers will use this feature out of context
+and create a support burden. However, using such platform type in conjunction with [capabilites selection](https://github.com/openshift/enhancements/blob/master/enhancements/installer/component-selection.md)
+could give us a clear signal about how to properly triage 3d party created replacements for Red Hat components.
+
+### Drawbacks
+
+Current approach with having statically defined platform list is already there, quite transparent and battle-hardened.
+Changing it by adding the new, way less specific platform type would mean a significant step away from current design patterns in this area.
+
+Also, future support strategy is not completely clear due to we have plans around enablement of 3d party platform-specific components, which
+codebase would be mostly out of our control.
+
+## Design Details
+
+### Open Questions
+
+- Should we explicitly communicate that the "External" platform is something that we do not support yet?
+- Should we gate the "External" platform addition behind the feature gate by generating separate CRD for TPNU clusters?
```

Comment:
What are the pros/cons? What benefits do we see of using a feature gate?

Reply:
Pros:
- No new api exposure during initial phases, no guarantees about its stability, lower possible mistakes cost, and so on.
- This compliments with "Start New Features as Tech Preview" policy

Cons:

- Mechanisms for doing so should be developed. The basic idea is there, but this would require developing respective tooling within "openshift/api".
- It will be harder to socialise within the community. People don't like "NoUpgrade" part.
- Testing upgrades between "y" streams would not be possible / way more complicated.

Can refer to this internal [doc](https://docs.google.com/document/d/1K2-zWyYQH-tHvPVwcKsPMxPJYNEOH5N3tENsBjDnqI0/edit#) for a bit more context.

Reply:
imo, if we can achieve the phase 2 goal of having our current operators behave as if "external" is equivalent to "none" platform, then i think we probably don't need the feature gate.